xikolo-ios/AbstractTabContentViewController.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  AbstractTabContentViewController.swift
+//  xikolo-ios
+//
+//  Created by Bjarne Sievers on 04.07.16.
+//  Copyright © 2016 HPI. All rights reserved.
+//
+
+import UIKit
+
+class AbstractTabContentViewController: UIViewController {
+
+    @IBOutlet var loginButton: UIBarButtonItem?
+
+    var isLoggedIn = UserProfileHelper.isLoggedIn()
+
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loginButton = self.navigationItem.rightBarButtonItem
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(AbstractTabContentViewController.updateLoginState), name: NotificationKeys.loginSuccessfulKey, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(AbstractTabContentViewController.updateLogoutState), name: NotificationKeys.logoutSuccessfulKey, object: nil)
+        updateUIAfterLoginLogoutAction()
+        self.navigationItem.hidesBackButton = true
+    }
+
+    override func shouldPerformSegueWithIdentifier(identifier: String, sender: AnyObject?) -> Bool {
+        switch identifier {
+        case "ShowLoginFromBarButton":
+            if isLoggedIn {
+                UserProfileHelper.logout() // TODO maybe fade transition
+                return false
+            } else {
+                return true
+            }
+        default:
+            return true
+        }
+    }
+
+    func updateLoginState() {
+        isLoggedIn = true
+        updateUIAfterLoginLogoutAction()
+    }
+
+    func updateLogoutState() {
+        isLoggedIn = false
+        updateUIAfterLoginLogoutAction()
+    }
+
+    func updateUIAfterLoginLogoutAction() {
+        if isLoggedIn {
+            dispatch_async(dispatch_get_main_queue(), {
+                self.navigationItem.rightBarButtonItem = nil
+
+            })
```

Comment:
Most of the time it's triggered by a notification and I had problems with it once, so I added the line. But i tested it now without it and it seems fine. If you're sure that it's not a problem, I will remove it.